Content.Server/Explosion/TriggerSystem.cs

```diff
@@ -121,5 +158,24 @@ public void HandleTimerTrigger(TimeSpan delay, IEntity triggered, IEntity? user
                 Trigger(triggered, user);
             });
         }
+
+        public void SetProximityFixture(EntityUid uid, TriggerOnProximityComponent component, bool remove)
+        {
+            var entity = EntityManager.GetEntity(uid);
+            var broadphase = Get<SharedBroadphaseSystem>();
+
+            if (entity.TryGetComponent(out PhysicsComponent? physics))
+            {
+                var fixture = physics.GetFixture(TriggerOnProximityComponent.FixtureID);
+                if (remove && fixture != null)
+                {
+                    broadphase.DestroyFixture(physics, fixture);
+                }
+                else
+                {
+                    broadphase.CreateFixture(physics, new Fixture(physics, component.Shape) { CollisionLayer = (int) (CollisionGroup.MobImpassable | CollisionGroup.SmallImpassable | CollisionGroup.SmallImpassable), Hard = false, ID = TriggerOnProximityComponent.FixtureID });
+                }
```

Comment:
sir we've been through this... this will create a new fixture when remove is true and there is no fixture present... just use what i sent you